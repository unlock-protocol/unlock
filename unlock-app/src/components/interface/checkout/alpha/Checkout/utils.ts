interface FetchDataOptions {
  recipients: string[]
  network: number
  lockAddress: string
}

export async function fetchData(url: string, options: FetchDataOptions) {
  try {
    const endpoint = new URL(url)
    endpoint.searchParams.append('network', options.network.toString())
    endpoint.searchParams.append('lockAddress', options.lockAddress)

    for (const recipient of options.recipients) {
      endpoint.searchParams.append('recipients[]', recipient)
    }

    const abortController = new AbortController()

    const timer = setTimeout(() => {
      abortController.abort()
    }, 5000)

    const response = await fetch(endpoint, {
      headers: {
        'Content-Type': 'application/json',
      },
      signal: abortController.signal,
    })

    clearTimeout(timer)

    if (!response.ok) {
      throw new Error('Failed to fetch data.')
    }

    const json = await response.json()
    return json.data as string[]
  } catch {
    return null
  }
}
