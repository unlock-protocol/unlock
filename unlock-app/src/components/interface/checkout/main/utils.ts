import { PaywallLockConfigType } from '@unlock-protocol/core'
import { PaywallConfig } from '~/unlockTypes'
interface FetchRecipientsDataOptions {
  recipients: string[]
  network: number
  lockAddress: string
}

export async function fetchRecipientsData(
  url: string,
  { lockAddress, network, recipients }: FetchRecipientsDataOptions
) {
  try {
    const result: string[] = []
    for (const recipient of recipients) {
      const dataEndpoint = new URL(url)

      dataEndpoint.searchParams.append('network', network.toString())
      dataEndpoint.searchParams.append('lockAddress', lockAddress)
      dataEndpoint.searchParams.append('recipient', recipient)

      // We need to proxy to avoid cors.
      const endpoint = new URL('/data', 'https://rpc.unlock-protocol.com')
      endpoint.searchParams.append('url', dataEndpoint.toString())

      const abortController = new AbortController()

      const timer = setTimeout(() => {
        abortController.abort()
      }, 5000)

      const response = await fetch(endpoint, {
        headers: {
          'Content-Type': 'application/json; charset=UTF-8',
        },
        signal: abortController.signal,
      })

      clearTimeout(timer)

      if (!response.ok) {
        throw new Error('Failed to fetch data.')
      }

      const json = await response.json()
      result.push(json.data)
      return result
    }
  } catch (error) {
    console.error(error)
    return
  }
}

export function formatFiatPriceFromCents(priceInCents: number, currency = '$') {
  const formatted = (priceInCents / 100).toLocaleString(undefined, {
    maximumFractionDigits: 2,
    minimumFractionDigits: 2,
  })
  return `${currency} ${formatted}`
}
interface Options {
  paywallConfig?: PaywallConfig
  lock?: PaywallLockConfigType
}

export const shouldSkip = ({ lock, paywallConfig }: Options) => {
  const maxRecipients = lock?.maxRecipients || paywallConfig?.maxRecipients
  const minRecipients = lock?.minRecipients || paywallConfig?.minRecipients
  const hasMaxRecipients = maxRecipients && maxRecipients > 1
  const hasMinRecipients = minRecipients && minRecipients > 1
  const skipQuantity = !(hasMaxRecipients || hasMinRecipients)

  const skip = lock?.skipRecipient || paywallConfig?.skipRecipient
  const collectsMetadadata =
    lock?.metadataInputs ||
    paywallConfig?.metadataInputs ||
    paywallConfig?.emailRequired ||
    lock?.emailRequired
  const skipRecipient = Boolean(skip && !collectsMetadadata)

  return {
    skipQuantity,
    skipRecipient,
  }
}
