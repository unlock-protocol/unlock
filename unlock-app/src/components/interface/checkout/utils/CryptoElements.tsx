// https://stripe.com/docs/crypto/quickstart

import { createContext, useEffect, useState, useRef, useContext } from 'react'

interface CryptoElementsContext {
  onramp: any
}

// ReactContext to simplify access of StripeOnramp object
const CryptoElementsContext = createContext<CryptoElementsContext | undefined>(
  undefined
)
CryptoElementsContext.displayName = 'CryptoElementsContext'

interface CryptoElementsProps {
  stripeOnramp: any
  children: React.ReactNode
}

export const CryptoElements = ({
  stripeOnramp,
  children,
}: CryptoElementsProps) => {
  const [ctx, setContext] = useState(() => ({
    onramp: null,
  }))

  useEffect(() => {
    let isMounted = true

    Promise.resolve(stripeOnramp).then((onramp) => {
      if (onramp && isMounted) {
        setContext((ctx) => (ctx.onramp ? ctx : { onramp }))
      }
    })

    return () => {
      isMounted = false
    }
  }, [stripeOnramp])

  return (
    <CryptoElementsContext.Provider value={ctx}>
      {children}
    </CryptoElementsContext.Provider>
  )
}

// React hook to get StripeOnramp from context
export const useStripeOnramp = () => {
  const context = useContext(CryptoElementsContext)
  return context?.onramp
}

// React element to render Onramp UI
const useOnrampSessionListener = (
  type: string,
  session: any,
  callback: any
) => {
  useEffect(() => {
    if (session && callback) {
      const listener = (e: any) => callback(e.payload)
      session.addEventListener(type, listener)
      return () => {
        session.removeEventListener(type, listener)
      }
    }
    return () => {}
  }, [session, callback, type])
}

interface OnrampElementProps {
  clientSecret: string
  appearance: any
  onChange: (payload: any) => void
}

export const OnrampElement = ({
  clientSecret,
  appearance,
  onChange,
  ...props
}: OnrampElementProps) => {
  const stripeOnramp = useStripeOnramp()
  const onrampElementRef = useRef<HTMLDivElement | null>(null)
  const [session, setSession] = useState()

  const appearanceJSON = JSON.stringify(appearance)
  useEffect(() => {
    const containerRef = onrampElementRef.current
    if (containerRef) {
      // NB: ideally we want to be able to hot swap/update onramp iframe
      // This currently results a flash if one needs to mint a new session when they need to udpate fixed transaction details
      containerRef.innerHTML = ''

      if (clientSecret && stripeOnramp) {
        setSession(
          stripeOnramp
            .createSession({
              clientSecret,
              appearance: appearanceJSON ? JSON.parse(appearanceJSON) : {},
            })
            .mount(containerRef)
        )
      }
    }
  }, [appearanceJSON, clientSecret, stripeOnramp])

  useOnrampSessionListener('onramp_session_updated', session, onChange)

  return <div {...props} ref={onrampElementRef}></div>
}
