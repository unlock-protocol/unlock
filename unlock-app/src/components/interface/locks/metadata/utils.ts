export interface Ticket {
  event_start_date?: string
  event_start_time?: string
  event_address?: string
  event_meeting_url?: string
}
export interface MetadataFormData {
  name: string
  description?: string
  external_url?: string
  youtube_url?: string
  animation_url?: string
  background_color?: string
  ticket: Ticket
  properties: Attribute[]
  levels: Attribute[]
  stats: Attribute[]
  [key: string]: any
}

export interface Attribute {
  display_type?: string
  max_value?: number
  trait_type: string
  value: string | number
}

export interface Metadata {
  name: string
  image?: string
  description?: string
  external_url?: string
  youtube_url?: string
  animation_url?: string
  background_color?: string
  attributes?: Attribute[]
  [key: string]: any
}

export function toFormData({
  name,
  description,
  attributes,
  animation_url,
  external_url,
  youtube_url,
  background_color,
}: Metadata) {
  const ticket = attributes
    ?.filter((item) => item.trait_type.startsWith('event_'))
    .reduce((item, { trait_type, value }) => {
      item[trait_type as keyof Ticket] = value as string
      return item
    }, {} as Ticket)

  const stats = attributes?.filter(
    (item) => item.display_type === 'number' && typeof item.value === 'number'
  )
  const levels = attributes?.filter(
    (item) => typeof item.value === 'number' && !item.display_type
  )
  const properties = attributes?.filter(
    (item) =>
      typeof item.value === 'string' &&
      !item.max_value &&
      !item.trait_type.startsWith('event_')
  )
  return {
    name,
    description,
    background_color,
    animation_url,
    external_url,
    youtube_url,
    ticket,
    levels,
    properties,
    stats,
  } as MetadataFormData
}
