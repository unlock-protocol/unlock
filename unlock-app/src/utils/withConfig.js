import React from 'react'
import configure from '../config'
import { WrongNetwork, MissingProvider } from '../components/creator/FatalError'
import { ETHEREUM_NETWORKS_NAMES } from '../constants'
import { connect } from 'react-redux'

/**
 * Function which creates higher order component with the config
 * Taken from https://reactjs.org/docs/context.html#consuming-context-with-a-hoc
 */

const isServer = typeof window === 'undefined'
const config = configure(global)
const ConfigContext = React.createContext(config)

// This function takes a component...
export function withConfig(Component) {
  // ...and returns another component...
  function componentWithConfig(props) {
    // ... and renders the wrapped component with the context config!
    // Notice that we pass through any additional props as well

    const { store: reduxStore, router } = props

    if (!isServer) {
      // Ensuring that we have at least a provider
      if (Object.keys(config.providers).length === 0) {
        return (<MissingProvider />)
      }

      // Ensuring that the provider is using the right network!
      if (router.route !== '/provider' && config.isRequiredNetwork && !config.isRequiredNetwork(reduxStore.getState().network.name)) {
        return (<WrongNetwork currentNetwork={ETHEREUM_NETWORKS_NAMES[reduxStore.getState().network.name][0]} requiredNetwork={config.requiredNetwork} />)
      }
    }

    return (
      <ConfigContext.Consumer>
        {config => <Component {...props} config={config} />}
      </ConfigContext.Consumer>
    )
  }

  function mapStateToProps (state) {
    return {
      account: state.network.account, // TODO change account to base level
      network: state.network,
    }
  }

  componentWithConfig.getInitialProps = async context => {
    return {
      ...(Component.getInitialProps ? await Component.getInitialProps(context) : {}),
    }
  }

  return connect(mapStateToProps)(componentWithConfig)
}
