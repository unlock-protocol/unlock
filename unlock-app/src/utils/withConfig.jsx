import React, { useContext } from 'react'

/**
 * Function which creates higher order component with the config
 * Taken from https://reactjs.org/docs/context.html#consuming-context-with-a-hoc
 */

export const ConfigContext = React.createContext()

/**
 * This creates an HOC from a component and injects the configuration.
 * It also triggers errors if constraints are not respected.
 * @param {*} Component
 */
export default function withConfig(Component) {
  function componentWithConfig(props) {
    return (
      <ConfigContext.Consumer>
        {(config) => <Component {...props} config={config} />}
      </ConfigContext.Consumer>
    )
  }

  return componentWithConfig
}

export function useConfig() {
  return useContext(ConfigContext)
}
