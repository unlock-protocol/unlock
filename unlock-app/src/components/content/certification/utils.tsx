import { Metadata } from '~/components/interface/locks/metadata/utils'

interface CertificationUrlProps {
  metadata?: Partial<Metadata>
  lockAddress: string
  network: string | number
  tokenId?: string | number
}

export const getCertificationPath = ({
  metadata,
  lockAddress,
  network,
  tokenId,
}: CertificationUrlProps): string => {
  const slug = metadata?.slug

  if (slug) {
    return `/certification?s=${slug}`
  }

  if (tokenId) {
    return `/certification?lockAddress=${lockAddress}&network=${network}&tokenId=${tokenId}`
  }

  return `/certification?lockAddress=${lockAddress}&network=${network}`
}
