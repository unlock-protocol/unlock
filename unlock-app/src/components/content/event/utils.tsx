import dayjs from 'dayjs'
import utc from 'dayjs/plugin/utc'
import timezone from 'dayjs/plugin/timezone'
import { Metadata } from '~/components/interface/locks/metadata/utils'

dayjs.extend(utc)
dayjs.extend(timezone)

export const getEventDate = (ticket: any): Date | null => {
  if (ticket?.event_start_date) {
    const timestamp = [ticket.event_start_date, ticket.event_start_time].join(
      ' '
    )
    const dayjsLocal = dayjs.tz(timestamp, ticket.event_timezone)
    return dayjsLocal.toDate()
  }

  return null
}

export const getEventEndDate = (ticket: any): Date | null => {
  if (ticket?.event_end_date) {
    const timestamp = [ticket.event_end_date, ticket.event_end_time].join(' ')

    const dayjsLocal = dayjs.tz(timestamp, ticket.event_timezone)
    return dayjsLocal.toDate()
  }

  return null
}

interface EventUrlProps {
  metadata?: Partial<Metadata>
  lockAddress: string
  network: string | number
}

export const getEventPath = ({
  metadata,
  lockAddress,
  network,
}: EventUrlProps): string => {
  const slug = metadata?.slug

  if (slug) {
    return `/event/${slug}`
  }
  return `/event?lockAddress=${lockAddress}&network=${network}`
}

export const getEventUrl = ({
  metadata,
  lockAddress,
  network,
}: EventUrlProps): string => {
  if (typeof window !== 'undefined' && window?.location?.origin) {
    return `${window.location.origin}${getEventPath({
      metadata,
      lockAddress,
      network,
    })}`
  }
  return getEventPath({
    metadata,
    lockAddress,
    network,
  })
}
