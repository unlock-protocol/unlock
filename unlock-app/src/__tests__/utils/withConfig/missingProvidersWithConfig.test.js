import React from 'react'
import renderer from 'react-test-renderer'
import { withConfig } from '../../../utils/withConfig'

const state = {
  account: {
    address: '0xdeadbeef',
    balance: '1000',
  },
  network: {
    name: 4,
  },
}

const store = {
  getState: jest.fn(() => state),
  dispatch: jest.fn(),
  subscribe: jest.fn(),
}

const Component = () =>
  <div>
    An unlock component
  </div>

const ComponentWithConfig = withConfig(Component)

jest.mock('../../../config', () => jest
  .fn()
  .mockImplementation(() => {
    return {
      providers: {},
      isRequiredNetwork: () => true,
      requiredNetwork: 'dev',
    }
  })
)

describe('withConfig High Order Component', () => {
  describe('when the current network is not in the list of required networks', () => {
    it('should show a message indicating that the network needs to be changed', () => {
      const tree = renderer
        .create(<ComponentWithConfig store={store} router={{ route: '/' }} />)
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
