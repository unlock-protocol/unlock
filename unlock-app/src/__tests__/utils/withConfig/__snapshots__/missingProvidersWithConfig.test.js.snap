// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withConfig High Order Component when the current network is not in the list of required networks should show a message indicating that the network needs to be changed 1`] = `
.c2 {
  background-color: var(--brand);
  height: 56px;
  width: 56px;
  border-radius: 50%;
}

.c2 > svg {
  fill: white;
}

.c8 {
  background-color: var(--grey);
  cursor: pointer;
  border-radius: 50%;
  height: 24px;
  width: 24px;
  display: grid;
  padding: 0;
  border: 0;
}

.c8 > svg {
  fill: white;
  height: 24px;
  width: 24px;
}

.c8:hover {
  background-color: var(--link);
}

.c8:hover > svg {
  fill: white;
}

.c11 {
  background-color: var(--grey);
  cursor: pointer;
  border-radius: 50%;
  height: 48px;
  width: 48px;
  display: grid;
  padding: 0;
  border: 0;
}

.c11 > svg {
  fill: white;
  height: 48px;
  width: 48px;
}

.c11:hover {
  background-color: var(--link);
}

.c11:hover > svg {
  fill: white;
}

.c9 {
  display: none;
  position: relative;
  z-index: 1000;
  white-space: nowrap;
  font-family: 'IBM Plex Sans','Helvetica Neue',Arial,sans-serif;
  font-weight: 400;
  font-size: 12px;
  top: 12px;
  left: 50%;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
}

.c7:hover .c9 {
  display: grid;
}

.c4 {
  display: grid;
  grid-gap: 16px;
  grid-template-columns: 1fr repeat(3,24px);
  grid-auto-flow: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 70px;
}

.c5 {
  color: var(--grey);
}

.c6 {
  display: grid;
  grid-gap: 16px;
  grid-template-columns: 1fr repeat(3,24px);
  grid-auto-flow: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
}

.c10 {
  display: none;
  height: 100%;
}

.c10 .c7 {
  background-color: var(--white);
  -webkit-transition: all 500ms cubic-bezier(0.165,0.84,0.44,1);
  transition: all 500ms cubic-bezier(0.165,0.84,0.44,1);
  opacity: 1;
  pointer-events: visible;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c10 .c7:nth-child(1) {
  grid-row: 0;
  pointer-events: none;
}

.c10 .c7:nth-child(2) {
  grid-row: 1;
  pointer-events: none;
  display: none;
}

.c10 .c7 > svg {
  fill: var(--grey);
}

.c10 .c7:hover > svg {
  fill: var(--grey);
}

.c12 {
  background-color: var(--white);
  position: absolute;
  left: 0;
  top: 70px;
  width: 100%;
  height: auto;
  z-index: var(--foreground);
  padding-bottom: 30px;
  display: none;
  grid-template-columns: 92px 92px 92px;
  grid-gap: 0px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-transition: all 500ms cubic-bezier(0.165,0.84,0.44,1);
  transition: all 500ms cubic-bezier(0.165,0.84,0.44,1);
  opacity: 0;
  pointer-events: none;
  top: 50px;
}

.c12 .c7 {
  margin: 24px;
}

.c12 .c7 small {
  display: block;
  color: var(--grey);
  text-align: center;
  top: 5px;
  width: 100%;
  text-align: center;
}

.c0 {
  display: grid;
  grid-template-columns: 1fr minmax(280px,4fr) 1fr;
}

.c1 {
  display: grid;
  -webkit-align-items: start;
  -webkit-box-align: start;
  -ms-flex-align: start;
  align-items: start;
  height: 24px;
}

.c3 {
  color: var(--darkgrey);
  display: grid;
  row-gap: 24px;
}

.c13 {
  display: grid;
  row-gap: 16px;
  -webkit-column-gap: 32px;
  column-gap: 32px;
  border: solid 1px var(--lightgrey);
  grid-template-columns: 72px;
  grid-auto-flow: column;
  border-radius: 4px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 32px;
  padding-bottom: 40px;
}

.c14 {
  width: 72px;
}

.c15 {
  display: grid;
  grid-gap: 16px;
}

.c15 > h1 {
  font-weight: bold;
  color: var(--red);
  margin: 0px;
  padding: 0px;
}

.c15 > p {
  margin: 0px;
  padding: 0px;
  font-size: 16px;
  color: var(--dimgrey);
}

@media only screen and (min-device-width:300px) and (max-device-width:736px) {
  .c4 {
    grid-template-columns: 1fr 48px;
    height: 35px;
  }
}

@media (max-width:600px) {
  .c6 {
    display: none;
  }
}

@media (max-width:600px) {
  .c10 {
    display: grid;
  }
}

@media (max-width:600px) {
  .c12 {
    display: grid;
  }
}

@media only screen and (min-device-width:300px) and (max-device-width:736px) {
  .c1 {
    display: none;
  }
}

@media only screen and (min-device-width:300px) and (max-device-width:736px) {
  .c16 {
    display: none;
  }
}

@media (max-width:500px) {
  .c13 {
    grid-template-columns: 1fr;
    grid-auto-flow: row;
    padding: 16px;
  }
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <div
        className="c2"
        size="56px"
      >
        <svg
          viewBox="0 0 56 56"
        >
          <title />
          <path
            d="M42.315 22.461h-1.862v-6.92h-6.919v6.92H22.48v-6.92h-6.918v6.92h-1.877v3.288h1.877v5.18c0 6.481 5.606 11.77 12.485 11.77 6.84 0 12.406-5.289 12.406-11.77v-5.18h1.862zm-8.78 8.468a5.515 5.515 0 0 1-5.488 5.567 5.583 5.583 0 0 1-5.567-5.567v-5.18h11.054z"
          />
        </svg>
      </div>
    </a>
  </div>
  <div
    className="c3"
  >
    <header
      className="c4"
    >
      <h1
        className="c5"
      >
        
      </h1>
      <div
        className="c6"
      >
        <a
          className="c7 c8"
          href="/about"
          onClick={[Function]}
        >
          <svg
            viewBox="0 0 24 24"
          >
            <title>
              About
            </title>
            <path
              clipRule="evenodd"
              d="M12 8.4A1.2 1.2 0 1 0 12 6a1.2 1.2 0 0 0 0 2.4zm0 9.6a1 1 0 0 0 1-1v-6a1 1 0 1 0-2 0v6a1 1 0 0 0 1 1z"
              fillRule="evenodd"
            />
          </svg>
          <small
            className="c9"
          >
            About
          </small>
        </a>
        <a
          className="c7 c8"
          href="/jobs"
          onClick={[Function]}
        >
          <svg
            name="Jobs"
            viewBox="0 0 24 24"
          >
            <title />
            <path
              clipRule="evenodd"
              d="M13 5h-2a2 2 0 0 0-2 2H7a2 2 0 0 0-2 2v1h14V9a2 2 0 0 0-2-2h-2a2 2 0 0 0-2-2zm1 2a1 1 0 0 0-1-1h-2a1 1 0 0 0-1 1h4zm-9 4h14v4a2 2 0 0 1-2 2H7a2 2 0 0 1-2-2v-4z"
              fillRule="evenodd"
            />
          </svg>
          <small
            className="c9"
          >
            Join us
          </small>
        </a>
        <a
          className="c7 c8"
          href="https://github.com/unlock-protocol/unlock"
          onClick={[Function]}
        >
          <svg
            name="Github"
            viewBox="0 0 24 24"
          >
            <title />
            <path
              d="M9.021 21.829c.007 1.065.017 2.171.017 2.171h6.136c0-.419.016-1.795.016-3.502 0-1.191-.397-1.97-.84-2.364 2.756-.316 5.65-1.396 5.65-6.3 0-1.394-.479-2.533-1.272-3.426.127-.323.552-1.62-.123-3.378 0 0-1.038-.344-3.4 1.308a11.554 11.554 0 0 0-3.099-.43c-1.053.006-2.11.147-3.098.43C6.644 4.686 5.604 5.03 5.604 5.03c-.673 1.757-.248 3.055-.12 3.378-.793.893-1.275 2.032-1.275 3.426 0 4.892 2.89 5.987 5.638 6.31-.354.319-.674.881-.786 1.706-.706.327-2.498.89-3.602-1.06 0 0-.654-1.226-1.896-1.316 0 0-1.209-.016-.085.776 0 0 .812.393 1.374 1.867 0 0 .726 2.483 4.17 1.712z"
            />
          </svg>
          <small
            className="c9"
          >
            Source Code
          </small>
        </a>
      </div>
      <div
        className="c10"
        onClick={[Function]}
      >
        <a
          className="c7 c11"
          href={null}
          onClick={[Function]}
          size="48px"
        >
          <svg
            viewBox="0 0 56 42"
          >
            <title>
              Bars
            </title>
            <path
              d="M0 6h56V0H0v6zm0 18h56v-6H0v6zm0 18h56v-6H0v6z"
            />
          </svg>
          
        </a>
        <a
          className="c7 c11"
          href={null}
          onClick={[Function]}
          size="48px"
        >
          <svg
            viewBox="0 0 58 32"
          >
            <title>
              Chevron Up
            </title>
            <path
              d="M3 29L29 5l26 24"
            />
          </svg>
          
        </a>
      </div>
      <div
        className="c12"
      >
        <a
          className="c7 c11"
          href="/about"
          onClick={[Function]}
          size="48px"
        >
          <svg
            viewBox="0 0 24 24"
          >
            <title>
              About
            </title>
            <path
              clipRule="evenodd"
              d="M12 8.4A1.2 1.2 0 1 0 12 6a1.2 1.2 0 0 0 0 2.4zm0 9.6a1 1 0 0 0 1-1v-6a1 1 0 1 0-2 0v6a1 1 0 0 0 1 1z"
              fillRule="evenodd"
            />
          </svg>
          <small
            className="c9"
          >
            About
          </small>
        </a>
        <a
          className="c7 c11"
          href="/jobs"
          onClick={[Function]}
          size="48px"
        >
          <svg
            name="Jobs"
            viewBox="0 0 24 24"
          >
            <title />
            <path
              clipRule="evenodd"
              d="M13 5h-2a2 2 0 0 0-2 2H7a2 2 0 0 0-2 2v1h14V9a2 2 0 0 0-2-2h-2a2 2 0 0 0-2-2zm1 2a1 1 0 0 0-1-1h-2a1 1 0 0 0-1 1h4zm-9 4h14v4a2 2 0 0 1-2 2H7a2 2 0 0 1-2-2v-4z"
              fillRule="evenodd"
            />
          </svg>
          <small
            className="c9"
          >
            Join us
          </small>
        </a>
        <a
          className="c7 c11"
          href="https://github.com/unlock-protocol/unlock"
          onClick={[Function]}
          size="48px"
        >
          <svg
            name="Github"
            viewBox="0 0 24 24"
          >
            <title />
            <path
              d="M9.021 21.829c.007 1.065.017 2.171.017 2.171h6.136c0-.419.016-1.795.016-3.502 0-1.191-.397-1.97-.84-2.364 2.756-.316 5.65-1.396 5.65-6.3 0-1.394-.479-2.533-1.272-3.426.127-.323.552-1.62-.123-3.378 0 0-1.038-.344-3.4 1.308a11.554 11.554 0 0 0-3.099-.43c-1.053.006-2.11.147-3.098.43C6.644 4.686 5.604 5.03 5.604 5.03c-.673 1.757-.248 3.055-.12 3.378-.793.893-1.275 2.032-1.275 3.426 0 4.892 2.89 5.987 5.638 6.31-.354.319-.674.881-.786 1.706-.706.327-2.498.89-3.602-1.06 0 0-.654-1.226-1.896-1.316 0 0-1.209-.016-.085.776 0 0 .812.393 1.374 1.867 0 0 .726 2.483 4.17 1.712z"
            />
          </svg>
          <small
            className="c9"
          >
            Source Code
          </small>
        </a>
      </div>
    </header>
    <section
      className="c13"
    >
      <img
        className="c14"
        src="/static/images/illustrations/wallet.svg"
      />
      <div
        className="c15"
      >
        <h1>
          Wallet missing
        </h1>
        <p>
          It looks like you’re using an incompatible browser or are missig a crypto wallet. If you’re using Chrome or Firefox you can install
           
          <a
            href="https://metamask.io/"
          >
            Metamask
          </a>
          .
        </p>
      </div>
    </section>
  </div>
  <div
    className="c16"
  />
</div>
`;
